### PR TITLE
[BD-46] feat: fixed SelectableBox responsiveness

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**
## Description

Fixed SelectableBox responsiveness.
[JIRA](https://openedx.atlassian.net/browse/PAR-815)

image

### Deploy Preview

[SelectableBox component](https://deploy-preview-1362--paragon-openedx.netlify.app/components/selectablebox/)

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [x] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [x] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [x] Were your changes tested in the `example` app?
* [x] Is there adequate test coverage for your changes?
* [x] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [x] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [x] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [x] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

File: src/SelectableBox/SelectableBoxSet.jsx [hunk 1]
```diff
@@ -59,6 +64,7 @@ SelectableBoxSet.propTypes = {
   columns: PropTypes.number,
   /** A class that is be appended to the base element. */
   className: PropTypes.string,
+  mobileColumns: PropTypes.number,
```
Comment (adamstankiewicz) [inline src/SelectableBox/SelectableBoxSet.jsx:67]: Let's add a comment above this prop to give it a description in the "Props API" table on the documentation website?

File: src/SelectableBox/SelectableBoxSet.jsx [hunk 2]
```diff
@@ -18,9 +21,11 @@ const SelectableBoxSet = React.forwardRef(({
   onChange,
   type,
   columns,
+  mobileColumns,
   className,
 }, ref) => {
   const inputType = getInputType('SelectableBoxSet', type);
+  const isExtraSmall = useMediaQuery({ maxWidth: breakpoints.extraSmall.maxWidth });
```
Comment (adamstankiewicz) [inline src/SelectableBox/SelectableBoxSet.jsx:28]: [clarification] Hmm, this wouldn't allow the consumer to customize the breakpoint as implemented. What if they needed to have it switch to the `mobileColumns` at a different breakpoint than `extraSmall` for example? Similarly, I'm wondering if there would be a need to customize different column counts for different breakpoints?

for example: extra small (1 column), medium (2 columns), extra large (3 columns).

**Review by adamstankiewicz — COMMENTED** [review]
Left a comment/suggestion about ensuring consumers have enough flexibility with the responsiveness of this component.

File: src/SelectableBox/SelectableBoxSet.jsx (hunk 1 above)
Comment (PKulkoRaccoonGang) [reply src/SelectableBox/SelectableBoxSet.jsx]: Removed

File: src/SelectableBox/SelectableBoxSet.jsx (hunk 2 above)
Comment (PKulkoRaccoonGang) [reply src/SelectableBox/SelectableBoxSet.jsx]: I reworked our solution for this problem. Now it is possible to configure adaptability as flexible as possible in Selectable Box. Check, please

File: src/SelectableBox/README.md [hunk 3]
```diff
@@ -32,13 +32,15 @@ As ``Checkbox``
   };
   
   const isInvalid = () => checkedCheeses.includes('swiss');
-
+  const isExtraSmall = useMediaQuery({maxWidth: breakpoints.extraSmall.maxWidth});
```
Comment (adamstankiewicz) [inline src/SelectableBox/README.md:35]: nit: add space between the `{` and `}`, (e.g., `useMediaQuery({ maxWidth: breakpoints.extraSmall.maxWidth })`)

Same with the other cases.

File: src/SelectableBox/README.md [hunk 4]
```diff
@@ -137,7 +141,7 @@ As ``Checkbox`` with ``isIndeterminate``
         type={type}
         onChange={handleChange}
         name="cheeses"
-        columns={3}
+        columns={isExtraSmall ? 1: 3}
```
Comment (adamstankiewicz) [inline src/SelectableBox/README.md:144]: nit: add space between `1` and the `:`

**Review by adamstankiewicz — DISMISSED** [review]
LGTM, with 2 small formatting nits. CI is failing, but rebasing on `master` should fix it.

File: src/SelectableBox/README.md (hunk 3 above)
Comment (PKulkoRaccoonGang) [reply src/SelectableBox/README.md]: Corrected

**Review by adamstankiewicz — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@PKulkoRaccoonGang 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/1362&entry.1484655318=https://github.com/PKulkoRaccoonGang&entry.752974735=2022-06-08+08:19&entry.1917517419=2022-06-22+05:31&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**